src/udev/udev-builtin-path_id.c

```diff
@@ -635,8 +636,24 @@ static int find_real_nvme_parent(sd_device *dev, sd_device **ret) {
                 return r;
 
         /* If the 'real parent' is (still) virtual, e.g. for nvmf disks, refuse to set ID_PATH. */
-        if (path_startswith(devpath, "/devices/virtual/"))
-                return -ENXIO;
+        if (path_startswith(devpath, "/devices/virtual/")) {
+		const char *syspath;
+		const char *addr, *fc_pn, *name, *nvme_pn;
+		struct sd_device *fcdev = NULL, *fc_parent = NULL;
+
+		sd_device_get_sysattr_value(nvme, "address", &addr);
+		nvme_pn = strrchr(addr, '-') + 1;
+
+		sd_device_new_from_syspath(&fc_parent, "/sys/class/fc_host");
+		FOREACH_DEVICE_CHILD_WITH_SUFFIX(fc_parent, child, name) {
+			fcdev = child;
```

Comment:
I do not think it is necessary to assign `fcdev`. How about the following?
```c
if (sd_device_get_sysattr_value(child, "port_name", &fc_pn) < 0)
        continue;

if (streq(fc_pn, nvme_pn)) {
        *ret = sd_device_ref(child);
        return 0;
}      
```

Reply:
I will re-work according  to your comments! Thanks!